library/psa_crypto.c

```diff
@@ -5524,8 +5524,8 @@ static void psa_des_set_key_parity(uint8_t *data, size_t data_size)
  * Note: Function allocates memory for *data buffer, so given *data should be
  *       always NULL.
  */
-#if defined(MBEDTLS_PSA_BUILTIN_KEY_TYPE_ECC_KEY_PAIR) || \
-    defined(MBEDTLS_PSA_BUILTIN_KEY_TYPE_ECC_PUBLIC_KEY) || \
+#if defined(PSA_WANT_KEY_TYPE_ECC_KEY_PAIR) || \
+    defined(PSA_WANT_KEY_TYPE_ECC_PUBLIC_KEY) || \
```

Comment:
I don't think that change is correct. This function calls a number of functions from ECP and Bignum. The way we know these are available is by guarding with `MBEDTLS_PSA_BUILTIN_KEY_TYPE_ECC_xxx`. In the absence of those guards, we'd need to explicitly guard with `defined(MBEDTLS_ECP_C)` and no actual progress would be made.

I think the current guards are correct here: they're consistent with the definition of the function.

Reply:
I want to call `psa_key_derivation_output_key`. Using ECC key types and accelerated key types, i.e MBEDTLS_PSA_ACCEL_KEY_TYPE_ECC_KEY_PAIR, so no buildin key types enabled.
And this function is preventing the calls to the PSA crypto driver calls.

There was a similar issue like this for validating tag length: 86679c7bd8856a1a89b4576344554123d8a42c24
I need both of these changes to be able to  use cryptocell acceleration on Mbed TLS 3.2.1.

Reply:
> In the absence of those guards, we'd need to explicitly guard with `defined(MBEDTLS_ECP_C)` and no actual progress would be made.

So, I was wrong about the last part: yes, we need an explicit `ECP_C` guard, but progress is still being made, as it allows more interesting configurations to build.